eng/packages/http-client-csharp/generator/Azure.Generator/src/AzureTypeFactory.cs

```diff
@@ -109,5 +113,41 @@ public override MethodBodyStatement SerializeJsonValue(Type valueType, ValueExpr
         /// <inheritdoc/>
         protected override ClientProvider CreateClientCore(InputClient inputClient)
             => AzureClientPlugin.Instance.IsAzureArm.Value ? base.CreateClientCore(InputClientTransformer.TransformInputClient(inputClient)) : base.CreateClientCore(inputClient);
+
+        internal ResourceDataProvider CreateResourceData(InputModelType inputModelType)
+        {
+            if (_resourceDataProvdierCache.TryGetValue(inputModelType, out var resourceDataProvider))
+            {
+                return resourceDataProvider;
+            }
+
+            resourceDataProvider = new ResourceDataProvider(inputModelType);
+            _resourceDataProvdierCache.Add(inputModelType, resourceDataProvider);
+            return resourceDataProvider;
+        }
+
+        /// <inheritdoc/>
+        protected override IReadOnlyList<TypeProvider> CreateSerializationsCore(InputType inputType, TypeProvider typeProvider)
+        {
+            if (inputType is InputModelType inputModel
+                && typeProvider is ModelProvider modelProvider
+                && AzureClientPlugin.Instance.OutputLibrary.IsResource(StringHelpers.ToCleanName(inputType.Name))
+                && inputModel.Usage.HasFlag(InputModelTypeUsage.Json))
```

Comment:
Wondering if this logic can be moved into a visitor.

Reply:
Given the visitor is applied after all the type providers actually created.
If we want to change an existing type provider, we need to visit all the references they have been used, which is not cumbersome.

Reply:
What I wrote above is concern about ModelProvider -> ResourceDataProvider.
For serialization, they might be fine since they are not referenced in other providers.
But then, do we want to treat them differently? 
And should the sub-plugin have the ability to handle the serialization differently?

Reply:
And we do need to handle serialization differently for Azure, such as the serialization for Resource.
For instance, https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/src/Resources/Generated/ResourceGroupResource.Serialization.cs

Reply:
I think having this in the type factory makes more sense actually.